test/e2e/common/networking.go

```diff
@@ -28,6 +29,46 @@ var _ = ginkgo.Describe("[sig-network] Networking", func() {
 
 	ginkgo.Describe("Granular Checks: Pods", func() {
 
+		checkNodeConnectivity := func(config *e2enetwork.NetworkingTestConfig, protocol string, port int) {
+			// breadth first poll to quickly estimate failure.
+			failedPodsByHost := map[string][]*v1.Pod{}
+			// First time, we'll quickly try all pods, breadth first.
+			for _, endpointPod := range config.EndpointPods {
+				framework.Logf("Breadth first check of %v on host %v...", endpointPod.Status.PodIP, endpointPod.Status.HostIP)
+				if err := config.DialFromTestContainer(protocol, endpointPod.Status.PodIP, port, 1, 0, sets.NewString(endpointPod.Name)); err != nil {
+					if _, ok := failedPodsByHost[endpointPod.Status.HostIP]; !ok {
+						failedPodsByHost[endpointPod.Status.HostIP] = []*v1.Pod{}
+					}
+					failedPodsByHost[endpointPod.Status.HostIP] = append(failedPodsByHost[endpointPod.Status.HostIP], endpointPod)
+					framework.Logf("...failed...will try again in next pass")
+				}
+			}
+			errors := []error{}
```

Comment:
this struct is aggregated later, so that the end resulting test , all errors are printed out together, and ratio  of polls that failed is available , so that you can clearly know wether a cluster is in catastrophic state or wether there's just a few problematic nodes